File: selenium-webdriver-rspec/spec/02_flight_spec.rb
load File.dirname(__FILE__) + '/../test_helper.rb'

describe "Select Flights" do
  include TestHelper

  before(:all) do
    @driver = $driver = Selenium::WebDriver.for(browser_type, browser_options)
    driver.manage().window().resize_to(1280, 720)
    driver.manage().window().move_to(30, 78)
    driver.get(site_url)
    
    login_page = LoginPage.new(driver)
    login_page.login("agileway", "testwise")
  end

  before(:each) do
    goto_page("/")
    sleep 1 # for some webdriver verson, it might not wait page loaded
  end

  after(:all) do
    driver.quit unless debugging?
  end

  it "[3] Return trip" do
    flight_page = FlightPage.new(driver)
    flight_page.select_trip_type("return")
    flight_page.select_depart_from("Sydney")
    flight_page.select_arrive_at("New York")

    flight_page.select_depart_day("02")
    flight_page.select_depart_month("May 2016")
    flight_page.select_return_day("04")
    flight_page.select_return_month("June 2016")
    flight_page.click_continue

    expect(page_text).to include("2016-05-02 Sydney to New York")
    expect(page_text).to include("2016-06-04 New York to Sydney")
  end

  it "[2] One-way trip" do
    flight_page = FlightPage.new(driver)
    flight_page.select_trip_type("oneway")
    flight_page.select_trip_type("return")
    flight_page.select_depart_from("Sydney")
    flight_page.select_arrive_at("New York")

    flight_page.select_depart_day("02")
    flight_page.select_depart_month("May 2016")
    flight_page.click_continue

    expect(page_text).to include("2016-05-02 Sydney to New York")
  end


end

File: selenium-webdriver-nodejs-mocha/spec/02_flight_spec.js
var webdriver = require('selenium-webdriver'),
    By = webdriver.By,
    until = webdriver.until;
var test = require('selenium-webdriver/testing');
var assert = require('assert');

var driver;
const timeOut = 15000;

String.prototype.contains = function(it) {
    return this.indexOf(it) != -1;
};

const chrome = require("selenium-webdriver/chrome");
var helper = require('../test_helper');

var FlightPage = require('../pages/flight_page.js')

test.describe('Flight', function() {

    test.before(function() {
        this.timeout(timeOut);
        driver = new webdriver.Builder().forBrowser(helper.browserType()).setChromeOptions(helper.chromeOptions()).build();
        driver.manage().window().setSize(1280, 720);
        driver.manage().window().setPosition(30, 78);

        driver.get('https://travel.agileway.net');
        helper.login(driver, "agileway", "testwise");
    });

    test.beforeEach(function() {
        this.timeout(timeOut);
        driver.get(helper.site_url());
    });

    test.after(function() {
        if (!helper.is_debugging()) {
            driver.quit();
        }
    });

    test.it('[3] Return trip', function() {
        this.timeout(timeOut);
        let flight_page = new FlightPage(driver);
        flight_page.selectTripType("return")
        flight_page.selectDepartFrom("Sydney")
        flight_page.selectArriveAt("New York")
        flight_page.selectDepartDay("02")
        flight_page.selectDepartMonth("May 2016")
        flight_page.selectReturnDay("04")
        flight_page.selectReturnMonth("June 2016")
        flight_page.clickContinue()
		
        driver.findElement(By.tagName("body")).getText().then(function(the_page_text) {
            assert(the_page_text.includes("2016-05-02 Sydney to New York"))
            assert(the_page_text.includes("2016-06-04 New York to Sydney"))
        });
    });

    test.it('[2] One-way trip', function() {
        this.timeout(timeOut);
        let flight_page = new FlightPage(driver);
        flight_page.selectTripType("oneway")
        flight_page.selectDepartFrom("New York")
        flight_page.selectArriveAt("Sydney")
        flight_page.selectDepartDay("02")
        flight_page.selectDepartMonth("May 2016")
        flight_page.clickContinue()

        driver.findElement(By.tagName("body")).getText().then(function(the_page_text) {
            assert(the_page_text.includes("2016-05-02 New York to Sydney"))
        });
    });


});
